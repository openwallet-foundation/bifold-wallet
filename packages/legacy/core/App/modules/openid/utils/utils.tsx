import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}
