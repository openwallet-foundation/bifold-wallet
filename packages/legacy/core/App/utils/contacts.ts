import { BasicMessageRecord, ConnectionRecord, CredentialExchangeRecord, ProofExchangeRecord } from '@credo-ts/core'

import { BifoldAgent } from './agent'

interface ConnectionWithMessages {
  conn: ConnectionRecord
  msgs: (BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord)[]
}

interface ConnectionWithLatestMessage {
  conn: ConnectionRecord
  latestMsg: BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
}

/**
 * Function to fetch contacts (connections) in order of latest chat message without using hooks
 * @param agent - Credo agent
 * @returns ConnectionRecord[] sorted by most recent message
 */
export const fetchContactsByLatestMessage = async (agent: BifoldAgent, connections: ConnectionRecord[]): Promise<ConnectionRecord[]> => {
  const connectionsWithMessages = await Promise.all<ConnectionWithMessages>(
    connections.map(
      async (conn: ConnectionRecord): Promise<ConnectionWithMessages> => ({
        conn,
        msgs: [
          ...(await agent.basicMessages.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.proofs.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.credentials.findAllByQuery({ connectionId: conn.id })),
        ],
      })
    )
  )

  const connectionsWithLatestMessage: ConnectionWithLatestMessage[] = connectionsWithMessages.map((pair) => {
    return {
      conn: pair.conn,
      latestMsg: pair.msgs.reduce(
        (acc, cur) => {
          const accDate = acc.updatedAt || acc.createdAt
          const curDate = cur.updatedAt || cur.createdAt
          return accDate > curDate ? acc : cur
        },
        // Initial value if no messages exist for this connection is a placeholder with the date the connection was created
        { createdAt: pair.conn.createdAt } as BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
      ),
    }
  })

  return connectionsWithLatestMessage
    .sort(
      (a, b) =>
        new Date(b.latestMsg.updatedAt || b.latestMsg.createdAt).valueOf() -
        new Date(a.latestMsg.updatedAt || a.latestMsg.createdAt).valueOf()
    )
    .map((pair) => pair.conn)
}
