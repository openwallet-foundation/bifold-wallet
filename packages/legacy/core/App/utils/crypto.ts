import argon2 from 'react-native-argon2'

export const hashPIN = async (PIN: string, salt: string): Promise<string> => {
  try {
    const result = await argon2(PIN, salt, {})
    const { rawHash } = result

    return rawHash
  } catch (error) {
    throw new Error(`Error generating hash for PIN ${String((error as Error)?.message ?? error)}`)
  }
}
