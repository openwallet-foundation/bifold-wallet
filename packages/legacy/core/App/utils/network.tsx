import TcpSocket from 'react-native-tcp-socket'

import pools from '../configs/ledgers/indy'
import { GenesisTransaction } from '../types/genesis'

export const canConnectToHost = async (host: { host: string; port: number }): Promise<boolean> =>
  new Promise((resolve) => {
    const socketTimeoutInMs = 3000
    const client = TcpSocket.createConnection(host, () => {
      resolve(true)
      client.destroy()
    })

    // Other events that can be safely be ignored. See the
    // library for more details:
    // https://www.npmjs.com/package/react-native-tcp-socket

    client.on('error', () => {
      client.destroy()
      resolve(false)
    })

    client.on('timeout', () => {
      client.destroy()
      client.removeAllListeners()

      resolve(false)
    })

    client.setTimeout(socketTimeoutInMs)
  })

export const fetchLedgerNodes = (indyNamespace = 'sovrin'): Array<{ host: string; port: number }> => {
  const [pool] = pools.filter((p) => p.indyNamespace === indyNamespace)
  if (!pool) {
    return []
  }

  const genesisTransactionsAsString = pool.genesisTransactions
  let genesisTransactions: Array<GenesisTransaction> = []

  if (genesisTransactionsAsString) {
    try {
      genesisTransactions = genesisTransactionsAsString.split('\n').map((g) => JSON.parse(g))
    } catch (error: unknown) {
      return []
    }
  }

  const nodes = genesisTransactions.map((g) => {
    return { host: g.txn.data.data.client_ip, port: parseInt(g.txn.data.data.client_port) }
  })

  return nodes
}
