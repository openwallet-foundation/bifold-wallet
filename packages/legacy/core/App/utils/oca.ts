import { CredentialExchangeRecord } from '@credo-ts/core'
import { AnonCredsProofRequestTemplatePayloadData, CredentialSharedProofData } from '@hyperledger/aries-bifold-verifier'
import {
  Attribute,
  CredentialOverlay,
  Field,
  OCABundleResolveAllParams,
  OCABundleResolverType,
  Predicate,
} from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../modules/openid/types'
import { BrandingOverlay } from '@hyperledger/aries-oca'

export const buildFieldsFromAnonCredsCredential = (credential: CredentialExchangeRecord): Array<Field> => {
  return credential?.credentialAttributes?.map((attr) => new Attribute(attr)) || []
}

export const buildFieldsFromW3cCredsCredential = (value: W3cCredentialDisplay): Array<Field> => {
  return (
    Object.entries(value.attributes).map(
      ([key, value]) =>
        new Attribute({
          name: key,
          value: value as string | number | null,
          mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
        })
    ) || []
  )
}

export const buildFieldsFromAnonCredsProofRequestTemplate = (
  data: AnonCredsProofRequestTemplatePayloadData
): Array<Field> => {
  const fields = []
  if (data.requestedAttributes) {
    for (const item of data.requestedAttributes) {
      if (item.name) {
        fields.push(new Attribute({ name: item.name, value: null, ...item }))
      }
      if (item.names) {
        for (const name of item.names) {
          fields.push(new Attribute({ name, value: null, ...item }))
        }
      }
    }
  }
  if (data.requestedPredicates) {
    for (const item of data.requestedPredicates) {
      fields.push(
        new Predicate({
          pType: item.predicateType,
          pValue: item.predicateValue,
          ...item,
        })
      )
    }
  }
  return fields
}

export const buildFieldsFromSharedAnonCredsProof = (data: CredentialSharedProofData): Array<Field> => {
  const fields = []
  for (const attribute of data.sharedAttributes) {
    fields.push(
      new Attribute({
        name: attribute.name,
        value: attribute.value,
      })
    )
  }
  for (const attributesGroup of data.sharedAttributeGroups) {
    for (const attribute of attributesGroup.attributes) {
      fields.push(
        new Attribute({
          name: attribute.name,
          value: attribute.value,
        })
      )
    }
  }
  for (const predicate of data.resolvedPredicates) {
    fields.push(
      new Predicate({
        name: predicate.name,
        pType: predicate.predicateType,
        pValue: predicate.predicateValue,
        satisfied: true,
      })
    )
  }
  return fields
}

export const buildOverlayFromW3cCredential = async ({
  credentialDisplay,
  language,
  resolver,
}: {
  credentialDisplay: W3cCredentialDisplay
  language: string
  resolver: OCABundleResolverType
}): Promise<CredentialOverlay<BrandingOverlay>> => {
  const params: OCABundleResolveAllParams = {
    identifiers: {
      schemaId: '',
      credentialDefinitionId: credentialDisplay.id,
    },
    meta: {
      alias: credentialDisplay.display.issuer.name,
      credConnectionId: undefined,
      credName: credentialDisplay.display.name,
    },
    attributes: buildFieldsFromW3cCredsCredential(credentialDisplay),
    language,
  }

  const bundle = await resolver.resolveAllBundles(params)

  return {
    ...(bundle as CredentialOverlay<BrandingOverlay>),
    presentationFields: buildFieldsFromW3cCredsCredential(credentialDisplay),
  }
}
