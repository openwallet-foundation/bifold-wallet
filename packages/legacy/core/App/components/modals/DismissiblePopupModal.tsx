import React from 'react'
import { useTranslation } from 'react-i18next'
import {
  StyleSheet,
  View,
  TouchableOpacity,
  TouchableWithoutFeedback,
  useWindowDimensions,
  ScrollView,
} from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { hitSlop } from '../../constants'
import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { testIdWithKey } from '../../utils/testable'
import Button, { ButtonType } from '../buttons/Button'
import { ThemedText } from '../texts/ThemedText'
import SafeAreaModal from './SafeAreaModal'

interface DismissiblePopupModalProps {
  title: string
  description?: string
  onCallToActionPressed?: GenericFn
  onCallToActionLabel?: string
  onDismissPressed: GenericFn
}

const DismissiblePopupModal: React.FC<DismissiblePopupModalProps> = ({
  title,
  description,
  onCallToActionPressed,
  onCallToActionLabel,
  onDismissPressed,
}) => {
  const { height, width, fontScale } = useWindowDimensions()
  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()
  const iconSize = 30

  const styles = StyleSheet.create({
    modalCenter: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: ColorPallet.notification.popupOverlay,
      padding: 20,
      minHeight: height,
      minWidth: width,
    },
    container: {
      backgroundColor: ColorPallet.notification.info,
      borderColor: ColorPallet.notification.infoBorder,
      borderRadius: 5,
      borderWidth: 1,
      padding: 10,
      minWidth: width - 2 * 25,
      maxWidth: width,
      flex: 1,
      maxHeight: fontScale < 1.7 ? '50%' : '70%',
    },
    headerContainer: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingHorizontal: 5,
      paddingTop: 5,
    },
    bodyContainer: {
      marginLeft: 10 + iconSize,
      paddingHorizontal: 5,
      paddingBottom: 5,
      flexShrink: 1,
      justifyContent: 'space-between',
    },
    headerTextContainer: {
      ...(fontScale < 1.7 ? { flexGrow: 1 } : { flexShrink: 1 }),
    },
    headerText: {
      ...TextTheme.bold,
      alignSelf: 'flex-start',
      color: ColorPallet.notification.infoText,
    },
    scrollViewContentContainer: {
      flexGrow: 1,
    },
    scrollViewStyle: {
      flex: 1,
    },
    bodyText: {
      ...TextTheme.normal,
      paddingVertical: 16,
      color: ColorPallet.notification.infoText,
    },
    footer: {
      paddingTop: 10,
    },
    infoIcon: {
      marginRight: 10,
      alignSelf: 'center',
    },
    dismissIcon: {
      alignSelf: fontScale < 1.7 ? 'flex-end' : 'flex-start',
    },
  })

  const infoIconName = 'info'
  const dismissIconName = 'clear'
  const iconColor = ColorPallet.notification.infoIcon

  return (
    <SafeAreaModal transparent>
      <TouchableOpacity onPress={onDismissPressed} accessible={false}>
        <View style={styles.modalCenter}>
          <TouchableWithoutFeedback accessible={false}>
            <View style={styles.container}>
              <View style={styles.headerContainer}>
                <View style={styles.infoIcon}>
                  <Icon name={infoIconName} size={iconSize} color={iconColor} />
                </View>
                <View style={styles.headerTextContainer}>
                  <ThemedText variant="bold" style={styles.headerText} testID={testIdWithKey('HeaderText')}>
                    {title}
                  </ThemedText>
                </View>
                <View style={styles.dismissIcon}>
                  <TouchableOpacity
                    onPress={onDismissPressed}
                    testID={testIdWithKey('Dismiss')}
                    accessibilityLabel={t('Global.Dismiss')}
                    accessibilityRole={'button'}
                    hitSlop={hitSlop}
                  >
                    <Icon name={dismissIconName} size={iconSize} color={iconColor} />
                  </TouchableOpacity>
                </View>
              </View>
              <View style={styles.bodyContainer}>
                <ScrollView
                  showsVerticalScrollIndicator={false}
                  contentContainerStyle={styles.scrollViewContentContainer}
                  style={styles.scrollViewStyle}
                >
                  <View onStartShouldSetResponder={() => true}>
                    <ThemedText selectable={true} style={styles.bodyText} testID={testIdWithKey('BodyText')}>
                      {description}
                    </ThemedText>
                  </View>
                </ScrollView>
                {onCallToActionPressed && (
                  <View style={styles.footer}>
                    <Button
                      title={onCallToActionLabel || t('Global.Okay')}
                      accessibilityLabel={onCallToActionLabel || t('Global.Okay')}
                      testID={testIdWithKey('Okay')}
                      buttonType={ButtonType.ModalPrimary}
                      onPress={onCallToActionPressed}
                    />
                  </View>
                )}
              </View>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </TouchableOpacity>
    </SafeAreaModal>
  )
}

export default DismissiblePopupModal
