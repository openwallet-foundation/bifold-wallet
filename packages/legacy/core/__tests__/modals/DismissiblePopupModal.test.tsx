import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import DismissiblePopupModal from '../../App/components/modals/DismissiblePopupModal'
import { testIdWithKey } from '../../App/utils/testable'

const title = 'Test Title'
const description = 'Lorem ipsum sit dolar amet'

describe('DismissiblePopupModal Component', () => {
  test('Renders correctly without call to action', () => {
    const onDismissPressed = jest.fn()
    const tree = render(
      <DismissiblePopupModal title={title} description={description} onDismissPressed={onDismissPressed} />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly with call to action', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const tree = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Triggers call to action on press', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const { getByTestId } = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )
    const okayButton = getByTestId(testIdWithKey('Okay'))
    fireEvent(okayButton, 'press')

    expect(onCallToActionPressed).toBeCalledTimes(1)
  })

  test('Triggers dismiss on press', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const { getByTestId } = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )
    const dismissButton = getByTestId(testIdWithKey('Dismiss'))
    fireEvent(dismissButton, 'press')

    expect(onDismissPressed).toBeCalledTimes(1)
  })
})
