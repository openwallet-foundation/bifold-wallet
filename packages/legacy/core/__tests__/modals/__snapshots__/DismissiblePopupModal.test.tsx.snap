// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DismissiblePopupModal Component Renders correctly with call to action 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      Object {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      Object {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={false}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 1334,
          "minWidth": 750,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "maxHeight": "70%",
            "maxWidth": 750,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "flexShrink": 1,
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "alignSelf": "flex-start",
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                ]
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Test Title
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            <View
              accessibilityLabel="Global.Dismiss"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 44,
                  "left": 44,
                  "right": 44,
                  "top": 44,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="com.ariesbifold:id/Dismiss"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#0099FF",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexShrink": 1,
              "justifyContent": "space-between",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            showsVerticalScrollIndicator={false}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                onStartShouldSetResponder={[Function]}
              >
                <Text
                  maxFontSizeMultiplier={2}
                  selectable={true}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                        "paddingVertical": 16,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Lorem ipsum sit dolar amet
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "paddingTop": 10,
              }
            }
          >
            <View
              accessibilityLabel="Global.Okay"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "opacity": 1,
                  "padding": 16,
                }
              }
              testID="com.ariesbifold:id/Okay"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        },
                        false,
                        false,
                        false,
                      ],
                    ]
                  }
                >
                  Global.Okay
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`DismissiblePopupModal Component Renders correctly without call to action 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      Object {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      Object {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={false}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 1334,
          "minWidth": 750,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "maxHeight": "70%",
            "maxWidth": 750,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "flexShrink": 1,
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "alignSelf": "flex-start",
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                ]
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Test Title
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            <View
              accessibilityLabel="Global.Dismiss"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 44,
                  "left": 44,
                  "right": 44,
                  "top": 44,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="com.ariesbifold:id/Dismiss"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#0099FF",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexShrink": 1,
              "justifyContent": "space-between",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "flexGrow": 1,
              }
            }
            showsVerticalScrollIndicator={false}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                onStartShouldSetResponder={[Function]}
              >
                <Text
                  maxFontSizeMultiplier={2}
                  selectable={true}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                        "paddingVertical": 16,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Lorem ipsum sit dolar amet
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
